src/papi-client.ts

```diff
@@ -0,0 +1,96 @@
+import {EventMessage} from "./models";
+import * as Promise from "bluebird";
+import {get, post, put, Response} from "superagent";
+
+export type BatchId = number;
+
+export interface PapiCredentials {
+    login: string;
+    password: string;
+}
+
+const DEFAUT_DELAY_MS = 1000;
+
+export class PapiClient {
+
+    public currentBatchId: BatchId;
+
+    constructor(
+        private apiUrl: string,
```

Comment:
done, everything is a string at the end but I don't want to validate given args ..